plugins/ut99/ut99.cpp

```diff
@@ -143,20 +143,14 @@ static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, floa
 	procptr_t cptr1 = peekProcPtr(cptr0 + 0x73C);
 	procptr_t cptr2 = peekProcPtr(cptr1 + 0x244);
 
-
-	wchar_t wservername[60];
-
-	ok = peekProc((procptr_t) cptr2, wservername);
-	if (!ok)
-		return false;
-
-	wservername[sizeof(wservername) / sizeof(wservername[0]) - 1] = '\0';
-
-	const std::string servername = utf16ToUtf8(wservername);
+	auto servername = peekProcVector< char16_t >(cptr2, 60);
+	if (servername.size() > 0) {
+		servername[servername.size() - 1] = '\0';
+	}
```

Comment:
Why are you explicitly setting the null byte?
And from my understanding with this approach you are overwriting the last character in case peekProc doesn't explicitly include a null-byte in the returned string itself

Reply:
> Why are you explicitly setting the null byte?

The plugin originally did that, I retained it to avoid a potential breakage (I didn't test the plugin as I don't own the game).

> And from my understanding with this approach you are overwriting the last character in case peekProc doesn't explicitly include a null-byte in the returned string itself

`peekProcVector` returns an `std::vector` with a size of 0 when the allocation fails.

In this case it should never fail because it's certain that the size we ask for is sane, but I wrote a check just in case.

As for the last character being replaced, I think the buffer inside the game is 60 bytes big, which means the last position is not usable anyway (reserved for the null character).

Reply:
Okay. Then maybe we should add a comment about what this is for to the code :point_up:

Reply:
I agree.

Reply:
I ended up using `std::u16string` instead, that way we don't have to handle it ourselves.